projects/components/src/load-async/loader/loader.component.ts

```diff
@@ -1,15 +1,38 @@
-import { ChangeDetectionStrategy, Component } from '@angular/core';
-import { IconType } from '@hypertrace/assets-library';
-import { IconSize } from '../../icon/icon-size';
+import { ChangeDetectionStrategy, Component, Input, OnChanges } from '@angular/core';
+import { ImagesAssetPath, LoaderTypes } from '@hypertrace/assets-library';
+import { assertUnreachable } from '@hypertrace/common';
 
 @Component({
   selector: 'ht-loader',
   styleUrls: ['./loader.component.scss'],
   template: `
     <div class="ht-loader">
-      <ht-icon icon="${IconType.Loading}" size="${IconSize.Large}"></ht-icon>
+      <img [ngClass]="[this.type]" [src]="this.imagePath" />
     </div>
   `,
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class LoaderComponent {}
+export class LoaderComponent implements OnChanges {
+  @Input()
+  public type: LoaderTypes = LoaderTypes.Horizontal;
+  public imagePath: ImagesAssetPath = ImagesAssetPath.LoaderHorizontal;
+
+  public ngOnChanges(): void {
+    // Unfortunatly passing undefined is overwriting default value
+    this.type = this.type ?? LoaderTypes.Horizontal;
```

Comment:
We shouldn't be modifying the input property. From where, is it getting the undefined value?

Reply:
updated